### PR TITLE
Automatic commit: Move 'cloudsdk-virtual-data-model-odata' from QA to…

### PR DESCRIPTION
**Pull request by jihoseo852**
Created Tutorials:
cloudsdk-virtual-data-model-odata/ QA link: https://developers-qa.sap.com/tutorials/cloudsdk-virtual-data-model-odata.html

cloudsdk-extensibility-type-safe-expand/ QA link: https://developers-qa.sap.com/tutorials/cloudsdk-extensibility-type-safe-expand.html

Updated the tutorials above.

### CODE REVIEW

**Comment by jihoseo852** [comment]
Applied all the changes.

I was told my Karsten, that this tutorial would be in a mission or a group with the mocking one. So this tutorial is the next step of the mocking-capabilities tutorial.

So the prerequisites are not 65minutes, but those are the steps that should be completed before starting this tutorial.

**Comment by jihoseo852** [comment]
Applied all the requested changes on cloudsdk-extensibility-type-safe-expand.

**Comment by MatKuhr** [comment]
@thecodester In case there is further feedback or questions please approach me via email.

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 1]
```diff
@@ -0,0 +1,506 @@
+---
+title: Extensibility, Type-safe Expand, and Dependency Injection with the Virtual Data Model for OData
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: Uppercase "Safe"

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+description: The following tutorial will introduce you to the latest features of the SAP Cloud SDK regarding extensibility, eager and type-safe expand as well as dependency injection with the Virtual Data Model for OData for any SAP S/4HANA system.
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: Too long -- just say "Learn ..."

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+primary_tag: topic>java
+---
+
+## Prerequisites
+ - [Introduce resilience to your application](https://developers.sap.com/tutorials/s4sdk-resilience.html)
+ - [Connect to OData Service on Cloud Foundry Using SAP Cloud SDK](s4sdk-odata-service-cloud-foundry)
+
+## Details
+### You will learn
+  - How you can use custom field extensions from S/4HANA within the virtual data model for OData.
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: All should be "How to ..." and no period at end

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+  - How you can join connected entities from the virtual data model in eager fashion.
+  - How to leverage dependency injection to decouple your client code better from the SDK-provided classes.
+
+---
+
+[ACCORDION-BEGIN [Step 1: ](Getting ready)]
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: No ing for step names -- Get ready

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+
+To successfully go through this tutorial, you will use the following `GetBusinessPartnerCommand` for all three features to the business partner API. The call itself requests the first and last name from all business partners which are customers.
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: ... who are customers

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 6]
```diff
@@ -0,0 +1,506 @@
...
+
+The file needs to be put under your `<projectroot>/application/src/main/java/com/sap/cloud/sdk/tutorial` directory:
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: For all such cases throughout tutorial, you should say "The following file ..." or "This file ..."

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 7]
```diff
@@ -0,0 +1,506 @@
...
+When you call this servlet on the server, you can see a result like this:
+
+!![Result](screenshot22.png)
+
+If you have a similar result, you are ready to proceed with this tutorial.
+
+
+[DONE]
+[ACCORDION-END]
+[ACCORDION-BEGIN [Step 2: ](Custom Field Extensibility)]
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: All step names are sentence case -- Custom field extensibility

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 8]
```diff
@@ -0,0 +1,506 @@
...
+
+#### What is custom field extensibility?
+Oftentimes, businesses require to enhance the standard data model of the SAP S/4HANA system. Using tools of the so-called In-App Extensibility concept, key users are able to introduce additional fields to certain data structures. As an application provider, this mechanism can be also used to introduce new data fields which are relevant to your application extension.
+
+Either way, you want to be able to consume and work with such custom fields in our application code.
+
+#### How-to
+In the example, the business partner data model has been enhanced with two custom fields to record by whom the business partner was originally proposed (field: `YY1_ProposedBy_bus`) and if the business partner was approved after proposal (field: `YY1_ApprovedBy_bus`).This modification is done by the so-called In-App extensibility capability of `SAP S/4HANA` which you can use based on your Fiori Launchpad with an authorized user.
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: Missing space before "This modification"

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 9]
```diff
@@ -0,0 +1,506 @@
...
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 3: ](Type-safe and Eager Expand)]
+
+#### What is type-safe and eager expand?
+So far, you have been working with the `BusinessPartner` entity only. However, this is merely the root entity of a more complex data model. For example, each `BusinessPartner` can be associated with zero-to-many `BusinessPartnerAddresses` which again can be associated with zero-to-many `BusinessPartnerEMailAddresses`. Another popular association in an ERP context are header-item relationships such as an invoice header and invoice line items.
+
+!![Tree](screenshot66.png)
+
+One possibility is to consider a lazy fetch of connected entities only using the `fetchXY()` methods that every instance exposes. In case of our business partner we could fetch the associated addresses with the following line of code:
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: You you and your, not we and our

File: tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md [hunk 10]
```diff
@@ -0,0 +1,506 @@
...
+            throw new ResilienceRuntimeException(e);
+        }
+    }
+}
+```
+
+The only two lines added to the initial example are:
+
+- `new BusinessPartnerField<String>("YY1_ApprovedBy_bus")`
+- `new BusinessPartnerField<String>("YY1_ProposedBy_bus"))`
```
Comment (thecodester) [inline tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md]: Did you mean to have 2 end parens?

**Comment by thecodester** [comment]
Please contact me via email and let me know where exactly to add these 2 new tutorials in which mission/group.

**Comment by MichaelCzcz** [comment]
Continued in https://github.com/SAPDocuments/Tutorials/pull/4706

**Pull request closed without merging**